### PR TITLE
[Example] Full finetuning/LoRA gpt-oss 20b/120b

### PR DESCRIPTION
**Pull request by Maknee**
Update docs with example for full finetuning/LoRA the gpt-oss models

Tested (run the relevant ones):

- [ ] Code formatting: install pre-commit (auto-check on commit) or `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `/smoke-test` (CI) or `pytest tests/test_smoke.py` (local)
- [ ] Relevant individual tests: `/smoke-test -k test_name` (CI) or `pytest tests/test_smoke.py::test_name` (local)
- [ ] Backward compatibility: `/quicktest-core` (CI) or `pytest tests/smoke_tests/test_backward_compat.py` (local)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 ----
 
 :fire: *News* :fire:
+- [Aug 2025] **OpenAI GPT-OSS models** (gpt-oss-120b, gpt-oss-20b) LoRA/Full Finetuning with 🤗 Accelerate on any cloud: [**example**](./llm/gpt-oss-sft/)
```
Comment (Michaelvll) [inline README.md]: How about combining this with the serving news below : )

File: llm/gpt-oss-sft/README.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
+# Run and Serve OpenAI gpt-oss Models with SkyPilot and vLLM
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]: The title is wrong?

File: llm/gpt-oss-sft/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+**Key Metrics to Monitor:**
+- **Loss**: Should decrease from ~2.23 to ~1.13 (indicating learning progress)
+- **Token Accuracy**: Should improve from ~54% to ~71% 
+- **Gradient Norm**: Should stabilize as training progresses
+- **Learning Rate**: Follows cosine schedule, decreasing over time
+
+**Expected Performance:**
+- Training time: ~5 minutes for 1 epoch on 2x8 H100 nodes
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]: Is this with infiniband on Nebius?

**Review by Michaelvll — COMMENTED** [review]
some minor comments

File: llm/gpt-oss-sft/README.md (hunk 3 above)
Comment (Maknee) [reply llm/gpt-oss-sft/README.md]: Yes

File: llm/gpt-oss-sft/README.md (hunk 2 above)
Comment (Maknee) [reply llm/gpt-oss-sft/README.md]: Will update

File: README.md (hunk 1 above)
Comment (Maknee) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -39,7 +39,7 @@
 ----
 
 :fire: *News* :fire:
-- [Aug 2025] Run and serve **OpenAI GPT-OSS models** (gpt-oss-120b, gpt-oss-20b) with one command on any infra: [**example**](./llm/gpt-oss/)
+- [Aug 2025] Run and serve **OpenAI GPT-OSS models** (gpt-oss-120b, gpt-oss-20b) with one command on any infra: [**example**](./llm/gpt-oss/) + LoRA/Full Finetuning with 🤗 Accelerate [**example**](./llm/gpt-oss-sft/)
```
Comment (Michaelvll) [inline README.md]:
```suggestion
- [Aug 2025] Serve and finetune **OpenAI GPT-OSS models** (gpt-oss-120b, gpt-oss-20b) with one command on any infra: [**serve**](./llm/gpt-oss/) + [**LoRA and full finetuning**](./llm/gpt-oss-sft/)
```

File: llm/gpt-oss-sft/README.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+```bash
+sky launch -c gpt-oss-120b-sft gpt-oss-120b-sft.yaml
+```
+
+### LoRA finetuning
+
+**For `gpt-oss-20b` with LoRA:**
+- Requirements: 1 node, 2x H100 GPU
+```bash
+sky launch -c zgpt-oss-20b-lora gpt-oss-20b-lora.yaml
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]:
```suggestion
sky launch -c gpt-oss-20b-lora gpt-oss-20b-lora.yaml
```

File: llm/gpt-oss-sft/README.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+![](https://i.imgur.com/TkoqCQK.png)
+
+On August 5, 2025, OpenAI released [gpt-oss](https://openai.com/open-models/), including two state-of-the-art open-weight language models: `gpt-oss-120b` and `gpt-oss-20b`. These models deliver strong real-world performance at low cost and are available under the flexible Apache 2.0 license.
+
+The `gpt-oss-120b` model achieves near-parity with OpenAI o4-mini on core reasoning benchmarks, while the `gpt-oss-20b` model delivers similar results to OpenAI o3-mini.
+
+This guide walks through how to finetune both models with LoRA/full finetuning using [🤗 Accelerate](https://github.com/huggingface/accelerate).
+
+![Cloud Logos](https://raw.githubusercontent.com/skypilot-org/skypilot/master/docs/source/images/cloud-logos-dark.png)
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]: may need to use the light version?

File: llm/gpt-oss-sft/README.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+```bash
+sky launch -c gpt-oss-20b-sft gpt-oss-20b-sft.yaml
+```
+
+**For `gpt-oss-120b` (larger model):**
+- Requirements: 4 nodes, 8x H200 GPUs each
+```bash
+sky launch -c gpt-oss-120b-sft gpt-oss-120b-sft.yaml
+```
+
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]: Can we inline the yaml here for better visibility?

File: llm/gpt-oss-sft/README.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+sky status
+
+# View logs
+sky logs <cluster-name>
+
+# Download results when complete
+sky down <cluster-name>
+```
+
+### Example training progress
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]:
```suggestion
### Example full finetuning progress
```

**Review by Michaelvll — COMMENTED** [review]
Thanks @Maknee for putting up this! Left some small comments.

File: llm/gpt-oss-sft/README.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
+# Finetuning OpenAI gpt-oss Models with SkyPilot
```
Comment (Michaelvll) [inline llm/gpt-oss-sft/README.md]: maybe we should use `gpt-oss-finetuning` as the folder name?

File: docs/source/examples/models/gpt-oss-sft.md [hunk 10]
```diff
@@ -0,0 +1 @@
+../../../../llm/gpt-oss-sft/README.md
```
Comment (Michaelvll) [inline docs/source/examples/models/gpt-oss-sft.md]: We can have it here instead: https://docs.skypilot.co/en/latest/examples/training/index.html

File: llm/gpt-oss-sft/README.md (hunk 7 above)
Comment (Maknee) [reply llm/gpt-oss-sft/README.md]: Updated

File: llm/gpt-oss-sft/README.md (hunk 6 above)
Comment (Maknee) [reply llm/gpt-oss-sft/README.md]: Updated

**Review by Michaelvll — APPROVED** [review]
Thanks @Maknee! LGTM
